actions/publish-github-pages/action.yml

```diff
@@ -1,10 +1,106 @@
+    description: 'custom CSS to insert into custom.scss'
+    required: false
+  debug:
+    description: 'whether to turn on debug printouts/functionality'
+    required: false
+  hugo_url:
+    description: 'URL of the Hugo site after deployment'
+    required: false
+  hugo_publish_dir:
+    description: 'directory in the repo that GitHub Pages will deploy to'
+    required: false
+  hugo_config:
+    description: 'location of to-be-written Hugo config file (actual location not important)'
+    required: false
+  htmltest_config:
+    description: 'location of to-be-written htmltest config file (actual location not important)'
+    required: false
+  github_token:
+    description: 'github token'
+    required: false
+  github_actor:
+    description: 'github actor'
+    required: false
+  github_pages_hugo_path:
+    description: 'location where the repo containing the Hugo files will be checked out to'
+    required: false
+  github_pages_pull_branch:
+    description: 'the branch of the repo which contains the documentation'
+    required: false
+  staging_dir:
+    description: 'directory to collate all docs and Hugo files in'
+    required: false
 runs:
-#  using: 'docker'
-#  image: 'Dockerfile'
   using: 'composite'
   steps:
-    - run: ${{ github.action_path }}/entrypoint.sh
+    - id: checkout-repos
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; checkout_repos()'
       shell: bash
-#    - run: ./actions/publish-github-pages/entrypoint.sh
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
+    - id: insert-custom-css
+      run: |
+           export CSS=${CSS:-$INPUT_CSS}
+           export GITHUB_PAGES_HUGO_PATH=${GITHUB_PAGES_HUGO_PATH:-$INPUT_GITHUB_PAGES_HUGO_PATH}
+           echo -e "$CSS" >> ${GITHUB_PAGES_HUGO_PATH}/themes/cloudposse/src/scss/custom.scss
+      shell: bash
+      env:
+        INPUT_CSS: ${{ inputs.CSS }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+    - id: collate-files
+      run: |
+           pip install GitPython pyyaml
+           python -c 'from checkout_and_collate import *; collate_files()'
+      shell: bash
+      env:
+        PYTHONPATH: ${{ github.action_path }}
+        GITHUB_PAGES_PUSH_PATH: ${{ github.action_path }}/${{ env.GITHUB_PAGES_DIRECTORY }}
+        INPUT_GITHUB_PAGES_HUGO_PATH: ${{ inputs.GITHUB_PAGES_HUGO_PATH }}
+        INPUT_GITHUB_PAGES_PULL_BRANCH: ${{ inputs.GITHUB_PAGES_PULL_BRANCH }}
+        INPUT_DEBUG: ${{ inputs.DEBUG }}
+        INPUT_CONTENT: ${{ inputs.CONTENT }}
+        INPUT_STAGING_DIR: ${{ inputs.STAGING_DIR }}
```

Comment:
What are the value of these `INPUT_` envs? When you just cast them to the other envs.... just move the exports up here and get rid of the intermediary envs.

Reply:
Unfortunately, the `inputs.*` variables get set to `''` whenever the action is called without the `with:` syntax. So, if we set `CONTENT: ${{ inputs.CONTENT }}`, it'll define `CONTENT=''`, whenever we use `env:` instead of `with:`, regardless of what value we give `CONTENT` in the `env:` section. Assigning the `with:` values to `INPUT_*` variables and then handling final assignment in bash or Python is my way of getting around that problem.

Reply:
Let's just double down on inputs then and deprecate the other ENVs.